codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala

```diff
@@ -147,19 +146,19 @@ object ModelBuilder {
 
   object TypeArgument {
 
-    def apply(fqn: FullyQualifiedName): TypeArgument =
-      apply(fqn.name, fqn.parent, fqn.descriptorObject)
+    def apply(messageType: ProtoMessageType): TypeArgument =
+      apply(messageType.name, messageType.parent, messageType.descriptorObject)
 
-    def apply(name: String, proto: PackageNaming, fqn: Option[FullyQualifiedName]): TypeArgument = {
+    def apply(name: String, proto: PackageNaming, messageTypeOpt: Option[ProtoMessageType]): TypeArgument = {
       if (name.nonEmpty && name.charAt(0).isLower) ScalarTypeArgument(ScalarType(name))
-      else MessageTypeArgument(FullyQualifiedName(name, name, proto, fqn))
+      else MessageTypeArgument(ProtoMessageType(name, name, proto, messageTypeOpt))
     }
   }
 
   /**
    * Type argument for Protobuf message types.
    */
-  case class MessageTypeArgument(fqn: FullyQualifiedName) extends TypeArgument
+  case class MessageTypeArgument(messageType: MessageType) extends TypeArgument
```

Comment:
`MessageTypeArgument` was taking a `FullyQualifiedName`, but now it takes `MessageType`. Kind of fits like a glove. :-)